packages/chat-core-aws-connect/src/infra/ChatCoreAwsConnectImpl.ts

```diff
@@ -0,0 +1,133 @@
+import { ChatCoreAwsConnect } from "../models/ChatCoreAwsConnect";
+import { MessageRequest, MessageResponse } from "@yext/chat-core";
+import { AwsConnectEvent } from "../models/AwsConnectEvent";
+import { EventCallback } from "../models/EventCallback";
+import "amazon-connect-chatjs";
+
+// TODO: Remove this type once the integration details are added to the MessageResponse type in chat-core
+type MessageResponseWithRegion = MessageResponse & {
+  integrationDetails: {
+    awsConnectHandoff: {
+      credentials: {
+        accessKeyId: string;
+        secretAccessKey: string;
+        sessionToken: string;
+      };
+      region: string;
+    };
+  };
+};
+
+/**
+ * The primary class for the chat-core integration with AWS Connect.
+ *
+ * @internal
+ */
+export class ChatCoreAwsConnectImpl implements ChatCoreAwsConnect {
+  private session?: connect.ActiveChatSession;
+  private eventListeners: Record<string, EventCallback[]> = {};
+
+  async init(messageRsp: MessageResponse): Promise<void> {
+    if (this.session) {
+      console.warn("Chat session already initialized");
+      return;
+    }
+
+    // TODO: Remove this type assertion once the integration details are added to the MessageResponse type in chat-core
+    const messageResponse = messageRsp as MessageResponseWithRegion;
+
+    const connectionCreds =
+      messageResponse.integrationDetails?.awsConnectHandoff?.credentials;
+    if (!connectionCreds) {
+      throw new Error(
+        "Integration credentials not specified. Cannot initialize chat session."
+      );
+    }
+
+    connect.ChatSession.setGlobalConfig({
+      loggerConfig: {
+        // There are five levels available - DEBUG, INFO, WARN, ERROR, ADVANCED_LOG. Default is INFO
+        level: connect.LogLevel.ERROR,
+        useDefaultLogger: true,
+      },
+      region: messageResponse.integrationDetails.awsConnectHandoff.region,
+    });
+
+    this.session = connect.ChatSession.create({
+      chatDetails: connectionCreds,
+      type: "CUSTOMER",
+    }) as connect.ActiveChatSession;
```

Comment:
`as connect.ActiveChatSession` shouldn't be necessary since we defined the type for `connect` already in `connect.d.ts`